### PR TITLE
Allow adding additional jar processors

### PR DESCRIPTION
**Pull request by natanfudge**

### CODE REVIEW

**Review by ramidzkh — CHANGES_REQUESTED** [review]
Doesn't let jar processors modify the final remapped jar. It is useful because e.g. remapped access wideners needs to be placed in the final mod archive

**Comment by natanfudge** [comment]
It doesn't allow processing mod jars as well which I would also like at some point but I would like to get this in soon.

File: src/main/java/net/fabricmc/loom/LoomGradleExtension.java [hunk 1]
```diff
@@ -67,6 +68,7 @@
 	private final ConfigurableFileCollection unmappedMods;
 
 	final List<LoomDecompiler> decompilers = new ArrayList<>();
+	final List<JarProcessor> jarProcessors = new ArrayList<>();
```
Comment (ramidzkh) [inline src/main/java/net/fabricmc/loom/LoomGradleExtension.java]: Why package private?

File: src/main/java/net/fabricmc/loom/LoomGradleExtension.java [hunk 2]
```diff
@@ -84,6 +86,10 @@ public void addDecompiler(LoomDecompiler decompiler) {
 		decompilers.add(decompiler);
 	}
 
+	public void addJarProcessor(JarProcessor processor) {
```
Comment (ramidzkh) [inline src/main/java/net/fabricmc/loom/LoomGradleExtension.java:96]: Document API methods?

File: src/main/java/net/fabricmc/loom/processors/JarProcessor.java [hunk 3]
```diff
@@ -31,7 +31,13 @@
 public interface JarProcessor {
```
Comment (ramidzkh) [inline src/main/java/net/fabricmc/loom/processors/JarProcessor.java:29]: https://github.com/FabricMC/fabric-loom/pull/225/files#r460656239

Comment (natanfudge) [reply src/main/java/net/fabricmc/loom/processors/JarProcessor.java:29]: ok

File: src/main/java/net/fabricmc/loom/LoomGradleExtension.java (hunk 2 above)
Comment (natanfudge) [reply src/main/java/net/fabricmc/loom/LoomGradleExtension.java:96]: ok

File: src/main/java/net/fabricmc/loom/LoomGradleExtension.java (hunk 1 above)
Comment (natanfudge) [reply src/main/java/net/fabricmc/loom/LoomGradleExtension.java]: I thought of using it in LoomGradlePlugin but that didn't go well

**Review by TheGlitch76 — APPROVED** [review]
not being able to process dependency jars or remapped jars is an issue, but that can be added later.
This works for 80% of use cases and imo defining the extended API would work better in a different PR anyways to avoid :bike::house:

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**